webapi/src/main/scala/org/knora/webapi/responders/admin/GroupsResponderADM.scala

```diff
@@ -299,8 +299,10 @@ class GroupsResponderADM(responderData: ResponderData) extends Responder(respond
                 case None => throw NotFoundException(s"Cannot create group inside project <${createRequest.project}>. The project was not found.")
             }
 
-            /* generate a new random group IRI */
-            groupIri = stringFormatter.makeRandomGroupIri(projectADM.shortcode)
+            // check group Iri
```

Comment:
Can you please add a comment here explaining the magic that is happening? The `id` being an `Optional`, is checked if `Some` and a new one is generated if `None`?

Reply:
The magic is happening inside the `checkEntityIri` that is used for formulating an Iri for any entity type (project, list, resource, etc). The description of `checkEntityIri` explains the magic: 

> Checks whether an entity with the provided custom IRI exists in the triplestore, if yes, throws an exception.
     * If no custom IRI was given, creates a random unused IRI.

Reply:
@subotic added a comment!